### ISSUE TITLE
Link to awesome-styled-components list

### ISSUE THREAD

**Issue by mxstbr** [styled-components/styled-components-website#123]
styled-components has such a great community behind it and there's so many modules built with it! We should link to https://github.com/styled-components/awesome-styled-components from the website to make sure people can discover them.

**Comment by SaraVieira** [comment]
@mxstbr Where do you think this should be linked ?

**Comment by mxstbr** [comment]
No idea!

**Comment by morajabi** [comment]
A community link in the navbar?

**Comment by mxstbr** [comment]
"Community" should, if anything, like to Spectrum, but what about "ecosystem"? We need to figure out what to put in the navbar anyway, see #115, so we can finally ship your awesome updates @morajabi!

**Comment by morajabi** [comment]
@mxstbr You're totally right, Max! I like "ecosystem". Also, it sounds like a proper link to put in out empty navbar :)

**Comment by SaraVieira** [comment]
And should that page simply have the contents of the readme in awesome styled components?

**Comment by mxstbr** [comment]
We either pull it in via the API or something (so we don't have to keep two lists up to date), or we just like to the repo.

**Comment by philpl** [comment]
We have a task tracking this: https://github.com/styled-components/styled-components-website/issues/99

I think having a generator API that pulls the readme from the awesome repo and transforms it would be optimal.

**Comment by morajabi** [comment]
Yeah, that's right. @mxstbr himself opened that too :)

**Comment by SaraVieira** [comment]
Yellow !

I'm giving this a try with the `md` component but I'm having some hardship :(

Is this the correct way to use it ?

```
import React, { Component } from 'react'
import DocsLayout from '../components/DocsLayout'
import { getReadme } from '../utils/githubApi'
import md from '../components/md'

class Ecosystem extends Component {
  constructor() {
    super()
    this.state = {
      loading: true,
      readme: null
    }
  }

  componentWillMount() {
    getReadme().then(readme =>
        this.setState({
          readme,
          loading: false
        })
    )
  }

  render() {
    return (
      <DocsLayout
        title="Ecosystem"
        description="Ecosystem of styled-components">
        {!this.state.loading && md(this.state.readme)}
      </DocsLayout>
    )
  }
}

export default Ecosystem
```

I will place the state stuff into a readme component but just wanted to get it working first

@mxstbr @morajabi @philpl

**Comment by morajabi** [comment]
@SaraVieira Sounds good! As we're using Next, It'd be great to have the data fetching (`getReadme()` in your snippet) stuff in `getInitialProps` so we'll fetch them on the server and you don't have to wait for it to finish sometimes.

**Comment by SaraVieira** [comment]
Sorry, here is the code 

```
import 'isomorphic-fetch'

export const getReadme = (repo = 'awesome-styled-components') =>
  fetch(`https://api.github.com/repos/styled-components/${repo}/readme`)
    .then(resp => resp.json())
    .then(data => atob(data.content))

```

But it doesn't work :(

```
md.js?96c69f0:56 Uncaught (in promise) TypeError: strings.join is not a function
    at md (md.js?96c69f0:56)
```

**Comment by morajabi** [comment]
Are you sure it's coming from this code block? Add a `catch` and check if it's this block.

**Comment by morajabi** [comment]
I think you should call `md``` instead of `md()`

**Comment by morajabi** [comment]
try calling md like this: `{!this.state.loading && md`${this.state.readme}`}`

**Comment by SaraVieira** [comment]
@morajabi Updated all the code to use isomorphic options and also placed the error stack

**Comment by morajabi** [comment]
Did this solve the issue?

**Comment by SaraVieira** [comment]
I tried that , that gives me a diffrent error:

```
Paragraph(...): A valid React element (or null) must be returned. You may have returned undefined, an array or some other invalid object.
Invariant Violation: Paragraph(...): A valid React element (or null) must be returned. You may have returned undefined, an array or some other invalid object.
    at invariant (http://localhost:3000/_next/1506977670642/commons.js:50:15)
    at ReactCompositeComponentWrapper.mountComponent (http://localhost:3000/_next/1506977670642/commons.js:30161:82)
    at mountComponent (http://localhost:3000/_next/1506977670642/commons.js:2391:35)
    at Object.module.exports._ReactReconciler2.default.mountComponent (http://localhost:3000/_next/1506977670642/commons.js:14891:35)
    at ReactDOMComponent.mountChildren (http://localhost:3000/_next/1506977670642/commons.js:34703:44)
    at ReactDOMComponent._createInitialChildren (http://localhost:3000/_next/1506977670642/commons.js:31691:32)
    at ReactDOMComponent.mountComponent (http://localhost:3000/_next/1506977670642/commons.js:31510:12)
    at mountComponent (http://localhost:3000/_next/1506977670642/commons.js:2391:35)
    at Object.module.exports._ReactReconciler2.default.mountComponent (http://localhost:3000/_next/1506977670642/commons.js:14891:35)
    at Object.updateChildren (http://localhost:3000/_next/1506977670642/commons.js:29896:51)
```

**Comment by SaraVieira** [comment]
It still gives me the same error above even with:

```
import React from 'react'
import DocsLayout from '../components/DocsLayout'
import { getReadme } from '../utils/githubApi'
import md from '../components/md'
import Loading from '../components/Loading'

const Ecosystem = ({readme}) => {
  console.log(typeof readme)
      return (<DocsLayout
        title="Ecosystem"
        description="Ecosystem of styled-components"
      >
        {typeof readme !== 'string' ? <Loading /> : md`${readme}`}
      </DocsLayout>)
}


Ecosystem.getInitialProps = async () => {
  const readme = await getReadme()

  return { readme }
}

export default Ecosystem

```
:(

**Comment by morajabi** [comment]
OK, push your changes so I can check them here.

**Comment by SaraVieira** [comment]
Here is the branch: https://github.com/styled-components/styled-components-website/tree/ecosystem-page

**Comment by morajabi** [comment]
@SaraVieira Maybe it's something in the readme that `md` cannot parse or needs it to be ecaped?

**Comment by morajabi** [comment]
OK! I got it working by this:
```
{typeof readme !== 'string' ? <Loading /> : md([readme])}
```

**Comment by morajabi** [comment]
Also, we'll need to remove some sections from the readme as they're not related to this page.

**Comment by philpl** [comment]
@morajabi @SaraVieira sorry for the hardship you two! You guys are right on track :)

Since it’s a tag function that’s supposed to be used for tagged template literals it accepts an array of strings and spreader arguments as interleaved placeholders. In this case the placeholders are expected to be React elements and the strings are expected to be markdown 😬

Feel free to modify the md tag function to just accept a string as a regular function as well 😬

For the page itself the most important thing to use is getInitialProps which will enable us to properly offload some things to SSR when the user enters the page directly. You can find some info on it on next’s repo

**Comment by morajabi** [comment]
@philpl 
> For the page itself the most important thing to use is getInitialProps which will enable us to properly offload some things to SSR when the user enters the page directly. You can find some info on it on next’s repo

Yeah, I told Sara above and she did that

> Feel free to modify the md tag function to just accept a string as a regular function as well 😬

I think just a type check (to check whether it's an array of strings or just a string) on strings would be fine.

**Comment by SaraVieira** [comment]
@morajabi You are the best , it works ❤️

@philpl No problem man , never apolagize for being busy :)

**Comment by morajabi** [comment]
I created a PR (#214) to fix that problem. After merging that, we'll be able to simply call `md` function like this: `md(readme)`. No more hacky-arrays :)

**Comment by SaraVieira** [comment]
@morajabi Nicce !

The page looks like this at the moment : https://styled-components-docs-hnzxcqefvt.now.sh/ecosystem ?

Watchu guys think ?

cc @mxstbr @philpl

**Comment by morajabi** [comment]
@SaraVieira As I said earlier:
> Also, we'll need to remove/change some sections from the readme after fetching as they're not related to this page.
Anyway, it's my opinion. What do you think @mxstbr @philpl?

**Comment by SaraVieira** [comment]
I know that why I deployed to get everyones feedback on what should be removed :)

**Comment by morajabi** [comment]
You're right, sorry for repeating obvious things :)

**Comment by SaraVieira** [comment]
@morajabi Don't apologize man , I know lots of people don't read comments so it's never bad to repeat stuff 😉

**Comment by philpl** [comment]
@SaraVieira Looks good! It'll need some formatting though. We should strip out the title (`# ...`), the table of content, the note, the badge, and the two footers.

The contribute section needs to be replaced with some content of our own that tells people to go to the repo.

And the last part is sth tricky. @morajabi The navbar and sidebar are pretty static and unconfigurable at the moment 😢  However, we'll need to include the navbar on the homepage eventually, and we'll need to grab all the headers (`## ...`) and generate an alternative sidebar for this page.

Is this possible at the moment or will it need loads of modifications?

**Comment by mxstbr** [comment]
This is awesome @SaraVieira! I'd definitely link to the repo from somewhere so folks can contribute their stuff to it?

**Comment by morajabi** [comment]
@philpl I can take a look at this tonight (GMT+3:30 🙃). Also if we closed #115 and shipped the nav bar before this, we need to add "Ecosystem" link to nav.

**Comment by SaraVieira** [comment]
@philpl I'll switch the API call to just get the raw one and see if it works :)

About the removing stuff aside from splitting the text in various places I don't know how we can achieve this 😢 

@mxstbr Sure ! I'll add links at the bottom 😄

**Comment by mxstbr** [comment]
> About the removing stuff aside from splitting the text in various places I don't know how we can achieve this 😢

Why not? It's not beautiful but if it works, it works!

**Comment by SaraVieira** [comment]
@mxstbr Just wanted to get that approval :p 

 > We should strip out the title (# ...), the table of content, the note, the badge, and the two footers.

I will do this then and have a PR today 😝

When is the next release ?

**Comment by mxstbr** [comment]
As soon as #115 lands, basically. That's blocking a release atm.

**Comment by morajabi** [comment]
@mxstbr Apart from this particular situation, I think a tool to generate an object tree from markdown and vice-versa can become handy in these places. Is there such a thing?

**Comment by philpl** [comment]
@morajabi Thanks! I don't think that the ecosystem page (+ the navbar fixes) will go in before #115. At least I hope not since it's a trivial PR I think 😆 Shall I open a new issue for you with the things mentioned above for the navbar/sidebar, regarding the homepage and ecosystem page?

**Comment by morajabi** [comment]
@philpl No problem!

**Comment by morajabi** [comment]
@philpl Can you make the issue with details so I can take a look?

**Comment by SaraVieira** [comment]
Closed by #99
